pkg/controller/agentregistration/server.go

```diff
@@ -74,9 +75,30 @@ func RunAgentRegistrationServer(ctx context.Context, port int, clientHolder *hel
 
 		// In the agent-registration case, the bootstrap sa is not created in the managed cluster namespace, because managed cluster is not created yet.
 		// Instead, it's in the pod namespace with the name "agent-registration-bootstrap".
-		bootstrapkubeconfig, _, err := bootstrap.CreateBootstrapKubeConfig(ctx, clientHolder, AgentRegistrationDefaultBootstrapSAName,
-			os.Getenv(constants.PodNamespaceEnvVarName),
-			7*24*3600, mergedKlusterletConfig)
+		ns := os.Getenv(constants.PodNamespaceEnvVarName)
+
+		var token []byte
+		if expirationStr == "" {
+			token, err = bootstrap.GetTokenFromSASecret(ctx, clientHolder.KubeClient, AgentRegistrationDefaultBootstrapSAName, ns)
```

Comment:
I think we have some behavior changes here, before the PR, when the expirationStr is empty, if getting the token from the sa failed, it will use the tokenRequest to create a new token, but now, it will just return fail. Is that on purpose?

Reply:
Yes, agent-registration is using a sa that always will always has the sa secret exists.

Reply:
This is because the agent-registration returns the manifests that may or may not applied on a cluster.

So I want the different URl matches a determined result:
* If the `expiration` is set, then token with that expiration.
* if the `expiration` is not set, then token has no expiration.

Reply:
What I want to avoid is: user doesn't set the `duration`, he get the result successfully, but now he doesn't know if the bootstraokubeconfig return by acm is long-term or 7 days.

He has to decode the token to know that.

But yes,  this is a behivior change comparing with the previous version.